qiita_db/support_files/patches/33.sql

```diff
@@ -0,0 +1,435 @@
+-- in the system and can be used to process an artifact
+CREATE TABLE qiita.software (
+    software_id          bigserial  NOT NULL,
+    name                 varchar  NOT NULL,
+    version              varchar  NOT NULL,
+    description          varchar  NOT NULL,
+    CONSTRAINT pk_software PRIMARY KEY ( software_id )
+ ) ;
+
+-- soft_command table - holds the information of a command in a given software
+-- this table should be renamed to command once the command table in the
+-- analysis table is merged with this one
+CREATE TABLE qiita.soft_command (
+    command_id           bigserial  NOT NULL,
+    name                 varchar  NOT NULL,
+    software_id          bigint  NOT NULL,
+    description          varchar  NOT NULL,
+    cli_cmd              varchar  ,
+    parameters_table     varchar  NOT NULL,
+    CONSTRAINT pk_soft_command PRIMARY KEY ( command_id )
+ ) ;
+CREATE INDEX idx_soft_command ON qiita.soft_command ( software_id ) ;
+ALTER TABLE qiita.soft_command ADD CONSTRAINT fk_soft_command_software FOREIGN KEY ( software_id ) REFERENCES qiita.software( software_id );
+
+-- Publication table - holds the minimum information for a given publication
+-- It is useful to keep track of the publication of the studies and the software
+-- used for processing artifacts
+CREATE TABLE qiita.publication (
+    doi                  varchar  NOT NULL,
+    pubmed_id            integer  ,
+    CONSTRAINT pk_publication PRIMARY KEY ( doi )
+ ) ;
+
+-- Software publictation table - relates each software package with the lists of
+-- its related publciations
+CREATE TABLE qiita.software_publication (
+    software_id          bigint  NOT NULL,
+    publication_doi      varchar  NOT NULL,
+    CONSTRAINT idx_software_publication_0 PRIMARY KEY ( software_id, publication_doi )
+ ) ;
+CREATE INDEX idx_software_publication_software ON qiita.software_publication ( software_id ) ;
+CREATE INDEX idx_software_publication_publication ON qiita.software_publication ( publication_doi ) ;
+ALTER TABLE qiita.software_publication ADD CONSTRAINT fk_software_publication FOREIGN KEY ( software_id ) REFERENCES qiita.software( software_id )    ;
+ALTER TABLE qiita.software_publication ADD CONSTRAINT fk_software_publication_0 FOREIGN KEY ( publication_doi ) REFERENCES qiita.publication( doi )    ;
+
+-- Add remaining FK to the artifact table. Creating here since the target tables
+-- do not exist when the artifact table was created
+ALTER TABLE qiita.artifact ADD CONSTRAINT fk_artifact_visibility FOREIGN KEY ( visibility_id ) REFERENCES qiita.visibility( visibility_id )    ;
+ALTER TABLE qiita.artifact ADD CONSTRAINT fk_artifact_soft_command FOREIGN KEY ( command_id ) REFERENCES qiita.soft_command( command_id )    ;
+
+-- Create a function to infer the visibility of the artifact from the
+-- raw data
+CREATE FUNCTION infer_rd_status(rd_id bigint, st_id bigint) RETURNS bigint AS $$
+    DECLARE
+        result bigint;
+    BEGIN
+        CREATE TEMP TABLE irds_temp
```

Comment:
I'm unsure on how to perform the functionality done in this function w/o using a temp table. Can you provide an alternative that executes the same functionality outlined here?